Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3112,6 +3168,39 @@ public void run() {
 		});
 	}
 
+	private void processRedeemRewardsResponse(final ServerResponse resp) {
+		boolean updateListener = false;
+		JSONObject requestObject = resp.getRequestObject();
+		if (requestObject != null) {
+			if (requestObject.has("bucket") && requestObject.has("amount")) {
+				try {
+					int redeemedCredits = requestObject.getInt("amount");
+					String creditBucket = requestObject.getString("bucket");
+					updateListener = redeemedCredits > 0;
+
+					int updatedCreditCount = prefHelper_.getCreditCount(creditBucket) - redeemedCredits;
+					updatedCreditCount = updatedCreditCount > 0 ? updatedCreditCount : 0;
+					prefHelper_.setCreditCount(creditBucket, updatedCreditCount);
+				} catch (JSONException e) {
+					e.printStackTrace();
+				}
+			}
+		}
+		final boolean finUpdateListener = updateListener;
+		Handler mainHandler = new Handler(context_.getMainLooper());
+		mainHandler.post(new Runnable() {
+			@Override
+			public void run() {
+				if (redeemStateChangedCallback_ != null) {
+					if (finUpdateListener)
```

Comment:
You have

```
if (x)
  doY(a, b)
else 
  doY(a, c)
```

You could consolidate that and avoid the duplication of that code block.

```
BranchError error = finUpdateListener ? null : BranchError...;
callback(finUpdateListener, error);
```